PolyEngine/Engine/Src/TransformComponent.cpp

```diff
@@ -0,0 +1,70 @@
+#include "TransformComponent.hpp"
+
+using namespace Poly;
+
+//-----------------------------------------------------------------------------
+void TransformComponent::SetParent(TransformComponent* parent) {
+	Parent = parent;
+	//TODO calculate new LocalTransformation based on parent GlobalTransformation
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalTranslation(const Vector& position) {
+	LocalTranslation = position;
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalRotation(const Quaternion& quaternion) { 
+	LocalRotation = quaternion; 
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+void TransformComponent::SetLocalScale(const Vector& scale) {
+	LocalScale = scale;
+	Dirty = true;
+}
+
+//------------------------------------------------------------------------------
+const Matrix& TransformComponent::GetLocalTransformationMatrix() const {
+	if (Dirty) {
+		Matrix translation;
+		Matrix rotation = LocalRotation.ToRotationMatrix();
+		Matrix scale;
+		translation.SetTranslation(LocalTranslation);
+		scale.SetScale(LocalScale);
+		LocalTransform = translation * rotation * scale;
+		Dirty = false;
+		UpdateGlobalTransformationCache();
+	}
+	return LocalTransform;
+}
+
+//------------------------------------------------------------------------------
+const Matrix& TransformComponent::GetGlobalTransformationMatrix() const {
+	UpdateGlobalTransformationCache();
+	return GlobalTransform;
+}
+
+//------------------------------------------------------------------------------
+const bool TransformComponent::UpdateGlobalTransformationCache() const {
+	if (Parent == nullptr) {
+		if (Dirty) {
```

Comment:
instead of the whole if write: ` const bool wasDirty = Dirty; GlobalTransform = GetLocalTransformationMatrix(); return wasDirty; `. Ofc not in a single line.